### PR TITLE
Section splitting for SO_Omni

### PR DESCRIPTION
**Pull request by alexandrak018**
Added the section splitting for the following spider: SO_Omni.

First of all, I think it is important to mention that I am being shown an odd number of decisions: 9k, while they should probably be around 5k. Therefore, I do not know how trustworthy my numbers are.
On top of this, I have the feeling there are decisions that are not on [Entscheidsuche](https://entscheidsuche.ch/). Why do I believe so it is because I searched, for example, for a decision from 15 December 2007 that did show up in my terminal, but it does not exist on the website under Solothurn.

**Results for DE language:**
- Header: 9293 / 9293 (100.00%) 
- Facts:  6923 / 9293 (74.50%) 
- Considerations: 7036 / 9293 (75.71%) 
- Rulings:        5703 / 9293 (61.37%) 
- Footer: 5740 / 9293 (61.77%)

**Remarks:**
There are mainly two types of "formats" for our decisions for this spider:

1.  [Type 1](https://entscheidsuche.ch/docs/SO_Omni/SO_OG_006_STBER-2015-33_2016-06-22.html )
2.  [Type 2](https://entscheidsuche.ch/docs/SO_Omni/SO_OG_004_ZKBER-2019-11_2019-04-04.html)

For _type 1_, the decisions have the tendency to be quite messy:
- I believe that most of them (actually, maybe even all of them), do not have "Rulings" or "Footer". Or if they do(I do not know this) they could probably be in the text, without a keyword that delimits them, such as "erkannt". But for me, even with a translator is too difficult to separate them. Therefore I recommend that someone who can speak German checks them.
- More, there are decisions like [this one](https://gerichtsentscheide.so.ch/cgi-bin/nph-omniscgi.exe?OmnisPlatform=WINDOWS&WebServerUrl=https://gerichtsentscheide.so.ch&WebServerScript=/cgi-bin/nph-omniscgi.exe&OmnisLibrary=JURISWEB&OmnisClass=rtFindinfoWebHtmlService&OmnisServer=7001&Parametername=WEB&Schema=JGWEB&Source=&Aufruf=getMarkupDocument&cSprache=DE&nF30_KEY=127731&W10_KEY=3581943&nTrefferzeile=50&Template=/simple/search_result_document.html) that do not have "keywords" to separate the sections, therefore it is hard for me to determine where the paragraphs belong to (FACTS, RULINGS, etc.)
-  Or there are also decisions that have very short paragraphs - [example](https://gerichtsentscheide.so.ch/cgi-bin/nph-omniscgi.exe?OmnisPlatform=WINDOWS&WebServerUrl=https://gerichtsentscheide.so.ch&WebServerScript=/cgi-bin/nph-omniscgi.exe&OmnisLibrary=JURISWEB&OmnisClass=rtFindinfoWebHtmlService&OmnisServer=7001&Parametername=WEB&Schema=JGWEB&Source=&Aufruf=getMarkupDocument&cSprache=DE&nF30_KEY=126549&W10_KEY=5224035&nTrefferzeile=48&Template=/simple/search_result_document.html)

For _type 2_, the decisions should be mostly fine as they are neatly written.

In conclusion, I did check a bunch of decisions and they seem to be splitted correctly. This, of course, for the ones I could. I can not promise the same for the "unusual" decisions.

### CODE REVIEW

**Comment by JoelNiklaus** [comment]
Thanks for the detailed explanations!

Could you please elaborate a bit more on the disparity of the number of decisions?

**Comment by alexandrak018** [comment]
To me, the total displays a number of 9293 decisions, but in reality, according to our Google Sheet or Entscheidsuche they should only be around 5000.
image

**Comment by Eldnn** [comment]
The problem was that some decisions were fetched twice from the website and had duplicated entries in the database. I have deleted the duplicated ones. 
However I cannot reproduce the problem of having decisions that are not on entscheidsuche.ch as I can't find any decision of the date 15 December 2007.

**Comment by alexandrak018** [comment]
Thank you very much, @Eldnn! Maybe I noted or read the decision's date wrong, therefore I apologize for any inconvenience I might have created!

So, I went through the decisions again and it seems like everything is fine now.
I was able to improve a tiny bit the results. There are decisions that are missing a section or multiple sections, and as we can observe from the results, they are mostly missing 'Rulings'.

**Updated results for DE language:**
- Header: 5589 / 5589 (100.00%) 
- Facts:  4460 / 5589 (79.80%) 
- Considerations: 4451 / 5589 (79.64%) 
- Rulings:        3581 / 5589 (64.07%) 
- Footer: 3592 / 5589 (64.27%)

**Comment by JoelNiklaus** [comment]
Great, thanks!
